bitbots_docs/docs/manual/software/coding_style.rst

```diff
@@ -0,0 +1,37 @@
+Coding Style
+============
+
+Because we value maintainability and readability of our codebase we have chosen to program according to the following
+style guides.
+
+Python
+------
+
+Our Python guide is based on `Google's style <http://google.github.io/styleguide/pyguide.html>`_ with the following
+adaptions:
+
+* **3.2** - Instead of an 80 character maximum line length we do 120 characters.
+* **3.8.2** - Since we have a _LICENSE_ file in every repository we refrain from including license boilerplate in every file.
+* as a docstring format we use `reST <https://www.python.org/dev/peps/pep-0287/>`_.
+
+
+C++
+---
+
+Our C++ guide is based on the `ROS C++ Style Guide <http://wiki.ros.org/CppStyleGuide>`_ except that our ``{`` are not
+on a new line.
+
+
+Git
+---
+
+We also have some conventions about how we want to use git. They are as follows:
+
+* Commits are written like ``when this commit is applied, it will <commit message>``.
+* Commits consist of a heading, newline and optional description.
+    * the heading is a short summary of the changes.
+    * the heading should start with a prefix like ``ci:``, ``walking:``, etc to show which part of the repository is affected.
+    * the description should not elaborate further on what was done but rather explain why something was done.
+* Any change should be done in a branch and then PRed to master.
+    * branch names should be prefixed with ``feature/``, ``fix/`` or none if not applicable.
```

Comment:
I tried to use 'refactor/...' for a branch name, which lead to the problem that the branch was not automatically created when I did git pull. So maybe the / is confusing git

Reply:
This should not be the case